### PR TITLE
Add doc and example dag for AWS Step Functions Operators

### PR DESCRIPTION
**Pull request by o-nikolas**
Part of a project to add, simplify and standardize AWS sample dags and docs in preparation for adding System Testing.

related: https://github.com/apache/airflow/pull/21828
related: https://github.com/apache/airflow/pull/22010
related: https://github.com/apache/airflow/pull/22533

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/amazon/aws/example_dags/example_step_functions.py
Comment (jedcunningham) [inline airflow/providers/amazon/aws/example_dags/example_step_functions.py:56]: Nit: for this simple case, I think I prefer `>>`.

**Review by jedcunningham — APPROVED** [review]

File: airflow/providers/amazon/aws/example_dags/example_step_functions.py
Comment (o-nikolas) [reply airflow/providers/amazon/aws/example_dags/example_step_functions.py:56]: Hey @jedcunningham,

We have been intentionally using `chain` for all our AWS example DAGs for consistency, rather than switching back and fourth.

With that in mind, do you still want to see this changed before you're willing to merge?

Comment (o-nikolas) [reply airflow/providers/amazon/aws/example_dags/example_step_functions.py:56]: ~~Hey @jedcunningham, just bumping this thread again, thanks!~~ Never mind, Jarek has just provided the second approval and merge :)

**Review by potiuk — APPROVED** [review]

File: airflow/providers/amazon/aws/example_dags/example_step_functions.py
Comment (jedcunningham) [reply airflow/providers/amazon/aws/example_dags/example_step_functions.py:56]: Sorry, yeah no objections here.

**Pull request merged**
